### PR TITLE
test: add xfail test for SavedModel restoring optimizer weights

### PR DESCRIPTION
**Pull request by adriangb**

### CODE REVIEW

**Comment by adriangb** [comment]
@bhack could you run kokoro please?

**Comment by bhack** [comment]
Sorry, I cannot assign labels in this repository

**Comment by qlzh727** [comment]
@k-w-w, can you take a look for this PR? Thanks.

**Comment by adriangb** [comment]
Hi folks, just checking in so this doesn't go stale. It'd be great to make some progress on this long-standing footgun

File: keras/saving/save_test.py [hunk 1]
```diff
@@ -385,9 +385,20 @@ def _assert_same_weights_and_metrics(self, model, loaded_model):
...
+        except (AssertionError, ValueError):
+          # Both errors can be raised, depending on the if shapes happen to be the same or not
+          return  # expected failure, but the test can't continue
+        else:
+          raise AssertionError(
+            "This test should have failed since SavedModel does not support restoring"
+            " optimizer weights"
+          )
+      else:
+        self.assertAllClose(model.optimizer.weights,
```
Comment (bhack) [inline keras/saving/save_test.py:400]: Honestly I would move this in a separate test function decorating it with https://github.com/tensorflow/tensorflow/blob/master/tensorflow/python/eager/def_function_test.py#L243 with a reference to the Github ticket. 

So when it is implemented the test will fail and we have a pointer to the original issue/feature request.
Then we could remove the decorator or simplify/unify the original test (the current method).

**Review by bhack — CHANGES_REQUESTED** [review]

**Comment by gbaned** [comment]
@adriangb Can you please resolve conflicts? Thank you!

**Comment by gbaned** [comment]
> I'd like someone to approve this / give me some indication that this won't just sit here for another 8 months before I spend the time to resolve merge conflicts.

@adriangb  Sorry for the delay, once conflicts resolved, we will review it quickly. Thank you!

File: keras/saving/save_test.py (hunk 1 above)
Comment (bhack) [reply keras/saving/save_test.py:400]: @adriangb Can you handle this comment?

Comment (adriangb) [reply keras/saving/save_test.py:400]: I don't think I'll have bandwidth anytime soon

Comment (bhack) [reply keras/saving/save_test.py:400]: No problem the comment itself is 6 months old

**Comment by gbaned** [comment]
Hi @adriangb Any update on this PR? Please. Thank you!

**Comment by gbaned** [comment]
Hi @adriangb I'm going to go ahead and close this PR, because it seems to have stalled. If you're still interested in pursing this (and responding to my comments), please feel free to reopen! Thank you!
